iio/iio_app/iio_app.c

```diff
@@ -342,21 +335,38 @@ int32_t iio_app_run(struct iio_app_device *devices, int32_t len)
 		}
 	}
 
+
 	iio_init_param.devs = iio_init_devs;
 	iio_init_param.nb_devs = len;
-	iio_init_param.nb_trigs = 0;
-	status = iio_init(&iio_desc, &iio_init_param);
+	iio_init_param.trigs = trigs;
+	iio_init_param.nb_trigs = nb_trigs;
+	status = iio_init(iio_desc, &iio_init_param);
 	if(status < 0)
 		goto error;
 
 	free(iio_init_devs);
 
 	do {
-		status = iio_step(iio_desc);
+		status = iio_step(*iio_desc);
 	} while (true);
 error:
 	status = print_uart_error_message(&uart_desc, uart_init_par, status);
 	return status;
 }
 
+int32_t iio_app_run(struct iio_app_device *devices, int32_t len)
+{
+	struct iio_desc	*iio_desc;
```

Comment:
We may not need to pass this along. I don't think we need it outside `iio_app_run`.